File: src/main/java/com/suifeng/web/italker/push/bean/db/UserFollow.java
package com.suifeng.web.italker.push.bean.db;

import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.GenericGenerator;
import org.hibernate.annotations.UpdateTimestamp;

import javax.persistence.*;
import java.time.LocalDateTime;

/**
 * 用户关系的Model
 * 用于用户之间进行还有关系的实现
 */
@Entity
@Table(name = "TB_USER_FOLLOW")
public class UserFollow {

    @Id
    @PrimaryKeyJoinColumn
    @GeneratedValue(generator = "uuid")
    @GenericGenerator(name = "uuid", strategy = "uuid2")
    @Column(updatable = false, nullable = false)
    private String id;

    //定义一个发起人
    //多对1 -> 你可以关注很多人，你的每一次关注都是一条记录
    //可以创建很多和关注的消息
    @ManyToOne(optional = false)    //不可选，必须存储，一条关注记录一定要有
    //定义关联的表字段名为originId，对应的是User.id
    @JoinColumn(name = "originId")
    private User origin;
    @Column(nullable = false, updatable = false, insertable = false)
    private String originId;

    //定义关注的目标
    @ManyToOne(optional = false)
    @JoinColumn(name = "targetId")
    private User target;
    @Column(nullable = false, updatable = false, insertable = false)
    private String targetId;

    //别名，也就是对target的备注名,别名可以为空
    @Column
    private String alias;

    @CreationTimestamp
    @Column(nullable = false)
    private LocalDateTime createAt = LocalDateTime.now();

    @UpdateTimestamp
    @Column(nullable = false)
    private LocalDateTime updateAt = LocalDateTime.now();

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public User getOrigin() {
        return origin;
    }

    public void setOrigin(User origin) {
        this.origin = origin;
    }

    public String getOriginId() {
        return originId;
    }

    public void setOriginId(String originId) {
        this.originId = originId;
    }

    public User getTarget() {
        return target;
    }

    public void setTarget(User target) {
        this.target = target;
    }

    public String getTargetId() {
        return targetId;
    }

    public void setTargetId(String targetId) {
        this.targetId = targetId;
    }

    public String getAlias() {
        return alias;
    }

    public void setAlias(String alias) {
        this.alias = alias;
    }

    public LocalDateTime getCreateAt() {
        return createAt;
    }

    public void setCreateAt(LocalDateTime createAt) {
        this.createAt = createAt;
    }

    public LocalDateTime getUpdateAt() {
        return updateAt;
    }

    public void setUpdateAt(LocalDateTime updateAt) {
        this.updateAt = updateAt;
    }
}

File: src/main/java/com/suifeng/web/italker/push/bean/db/UserFollow.kt
package com.suifeng.web.italker.push.bean.db

import org.hibernate.annotations.CreationTimestamp
import org.hibernate.annotations.GenericGenerator
import org.hibernate.annotations.UpdateTimestamp

import javax.persistence.*
import java.time.LocalDateTime

/**
 * 用户关系的Model
 * 用于用户之间进行还有关系的实现
 */
@Entity
@Table(name = "TB_USER_FOLLOW")
class UserFollow {

    @Id
    @PrimaryKeyJoinColumn
    @GeneratedValue(generator = "uuid")
    @GenericGenerator(name = "uuid", strategy = "uuid2")
    @Column(updatable = false, nullable = false)
    var id: String? = null

    //定义一个发起人
    //多对1 -> 你可以关注很多人，你的每一次关注都是一条记录
    //可以创建很多和关注的消息
    @ManyToOne(optional = false)    //不可选，必须存储，一条关注记录一定要有
    //定义关联的表字段名为originId，对应的是User.id
    @JoinColumn(name = "originId")
    var origin: User? = null
    @Column(nullable = false, updatable = false, insertable = false)
    var originId: String? = null

    //定义关注的目标
    @ManyToOne(optional = false)
    @JoinColumn(name = "targetId")
    var target: User? = null
    @Column(nullable = false, updatable = false, insertable = false)
    var targetId: String? = null

    //别名，也就是对target的备注名,别名可以为空
    @Column
    var alias: String? = null

    @CreationTimestamp
    @Column(nullable = false)
    var createAt = LocalDateTime.now()

    @UpdateTimestamp
    @Column(nullable = false)
    var updateAt = LocalDateTime.now()
}
